### PR TITLE
388 - display food content in demo app

### PR DESCRIPTION
**Pull request by florentmaitre**

### CODE REVIEW

File: demo/src/main/java/com/orange/ods/demo/ui/components/radiobuttons/ComponentRadioButtons.kt [hunk 1]
```diff
@@ -44,28 +44,17 @@ fun ComponentRadioButtons() {
                 .verticalScroll(rememberScrollState())
                 .padding(bottom = dimensionResource(id = R.dimen.spacing_m))
         ) {
-            val selectedRadio = remember { mutableStateOf(R.string.component_element_item1) }
+            val recipes = LocalRecipes.current.take(3)
+            val selectedRadio = remember { mutableStateOf(recipes.firstOrNull()?.title) }
```
Comment (paulinea) [inline demo/src/main/java/com/orange/ods/demo/ui/components/radiobuttons/ComponentRadioButtons.kt:48]: I think this should be a `rememberSaveable` cause we lost the selection on screen orientation change.

File: demo/src/main/java/com/orange/ods/demo/ui/components/switches/ComponentSwitches.kt [hunk 2]
```diff
@@ -42,7 +45,7 @@ fun ComponentSwitches() {
                 .padding(bottom = dimensionResource(id = R.dimen.spacing_m))
         ) {
             SwitchListItem(
```
Comment (paulinea) [inline demo/src/main/java/com/orange/ods/demo/ui/components/switches/ComponentSwitches.kt:47]: Should we add a list as for radio buttons?

File: demo/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -282,12 +278,12 @@
...
-    <string name="component_tab_alerts">Alerts</string>
-    <string name="component_tab_calendar">Calendar</string>
-    <string name="component_tab_account">Account</string>
-    <string name="component_tab_settings">Settings</string>
+    <string name="component_tab_coffee">Coffee</string>
+    <string name="component_tab_cooking_pot">Cooking Pot</string>
+    <string name="component_tab_ice_cream">Ice Cream</string>
+    <string name="component_tab_restaurant">Restaurant</string>
+    <string name="component_tab_favorites">Favorites</string>
+    <string name="component_tab_information">Information</string>
```
Comment (paulinea) [inline demo/src/main/res/values/strings.xml:286]: Should we mutualize these data used by tabs and bottom navigation components by creating a data class with a title and a drawable resource?

Comment (florentmaitre) [reply demo/src/main/res/values/strings.xml:286]: I created a `NavigationItem` enum used by both tabs and bottom navigation. I could not find a better name, another solution would be to rename it to `BottomNavigationItem` and add a typealias:

 ```kotlin
typealias TabItem = BottomNavigationItem
```

File: demo/src/main/java/com/orange/ods/demo/data/recipes/RecipesRepositoryImpl.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    private val foodParser = RecipesParser()
+
+    override fun getRecipes(): Flow<List<Recipe>> {
+        return flow {
+            try {
+                val jsonString = context.resources.openRawResource(R.raw.recipes).contentAsString().orEmpty()
+                val recipes = foodParser.parseRecipes(jsonString)
+                emit(recipes)
+            } catch (_: Exception) {
```
[the remark this answers is not in the document]
Comment (paulinea) [reply demo/src/main/java/com/orange/ods/demo/data/recipes/RecipesRepositoryImpl.kt:32]: I think we can no?

File: demo/src/main/res/values/strings.xml (hunk 3 above)
Comment (paulinea) [reply demo/src/main/res/values/strings.xml]: I agree for type aliases. Let's keep `NavigationItem` for now.

**Review by paulinea — APPROVED** [review]

**Pull request merged**